flow-model-generator/modules/flow-model-generator-ls-extension/src/main/java/io/ballerina/flowmodelgenerator/extension/FlowModelGeneratorService.java

```diff
@@ -435,6 +439,28 @@ public CompletableFuture<FlowModelSourceGeneratorResponse> addErrorHandler(FileP
         });
     }
 
+    @JsonRequest
+    public CompletableFuture<IsNodeUsedResponse> isNodeUsed(IsNodeUsedRequest request) {
+        return CompletableFuture.supplyAsync(() -> {
+           IsNodeUsedResponse response = new IsNodeUsedResponse();
+           try {
+               Path filePath = Path.of(request.filePath());
+               Optional<SemanticModel> semanticModel = this.workspaceManager.semanticModel(filePath);
+               Optional<Document> document = this.workspaceManager.document(filePath);
+               if (semanticModel.isEmpty() || document.isEmpty()) {
+                   response.setUsed(false);
+                   return response;
+               }
+               BasicNodeGenerator basicNodeGenerator = new BasicNodeGenerator(request.flowNode());
+               List<Location> references = semanticModel.get().references(document.get(), basicNodeGenerator.getNodeStartLocation());
+               response.setUsed(!references.isEmpty());
```

Comment:
I think we have to check the size is grater than 1. Because this references api returns the symbol declaration place as well.